### DISCUSSION TITLE
Incompatibility with later ES versions

### DISCUSSION THREAD

**Discussion by TomDo1234** [payloadcms/payload#1827] [Feature Requests & Ideas]
# Bug Report

I am using the default tsconfig 
`{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "strict": false,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "outDir": "./dist",
    "rootDir": "./src",
    "jsx": "react",
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules",
    "dist",
    "build",
  ],
  "ts-node": {
    "transpileOnly": true
  }
}
`

The issue with ES5 is that there is no top level await. I wanted top level await so I tried to increase to es2022, Had to then add moduleResolution: Node, then module: ES2022 in order to stop the import errors from appearing and get the top level await to not have an error on my IDE. Then I got an error in the console when trying to build, had to do add type: module to package.json.

After that I run yarn build, it does build but when you run it: import error on [Error [ERR_REQUIRE_ESM]: require() of ES Module not supported](https://stackoverflow.com/questions/69081410/error-err-require-esm-require-of-es-module-not-supported)

## Steps to Reproduce

1. Try to use top level await on your Payload project
2. Get an error, VS code will recommend you setting module to es2022 or higher, and target to ES2017 or higher. Do both of those
3. Observe errors on your import statements, hover over them and you will see a recommendation to add module resolution Node to the TSCONFIG.
4. After you do that, run yarn build, you will get an error telling you to add type module to package.json
5. Build it, it may build correctly but if you do yarn serve you will get a REQUIRE_ESM error.

## Other Details

Latest version of Payload as of the creation date of this post. 1.5.3

**Comment by jmikrut** [comment]
Hey @TomDo1234 — I do think that we should audit our package dependencies for any that are incompatible with ESM, although this is more of a future feature rather than a current issue. I am also extremely jaded by the huge schism that ESM vs. CommonJS / etc. have caused in the Node ecosystem. Honestly I need to set some time aside to really dig in to find out if we can even support both seamlessly.

Would you be interested in helping evaluate what's causing these `REQUIRE_ESM` issues? Otherwise I'm sure someone on my team can get to it at some point but this will likely be a priority 2 or 3 item for us I'm afraid.

Going to convert to a discussion, but let's keep conversation going there!

**Reply by TomDoFuture** [reply]
Yeah sure Jmi, This weekend I will be sure to inspect the source to see what exactly I can do to fix the dependencies.

The use case btw is that I want to use await expressions in the payload config and the server.ts to LOAD in credentials dynamically from a third party secure service like AWS SSM Params store.
